backtrace-library/src/main/java/backtraceio/library/models/BacktraceData.java

```diff
@@ -98,103 +97,224 @@ public class BacktraceData {
-                clientAttributes);
-        this.attributes = backtraceAttributes.attributes;
+    public String getSymbolication() {
+        return symbolication;
+    }
 
-        setAnnotations(backtraceAttributes.getComplexAttributes());
+    public long getTimestamp() {
+        return timestamp;
     }
 
-    /**
-     * Set report information such as report identifier (UUID), timestamp, classifier
-     */
-    private void setReportInformation() {
-        BacktraceLogger.d(LOG_TAG, "Setting report information");
-        uuid = report.uuid.toString();
-        timestamp = report.timestamp;
-        classifiers = report.exceptionTypeReport ? new String[]{report.classifier} : null;
-        langVersion = System.getProperty("java.version"); //TODO: Fix problem with read Java version
-        agentVersion = BacktraceClient.version;
+    public String getLangVersion() {
+        return langVersion;
     }
 
-    /**
-     * Set information about all threads
-     */
-    private void setThreadsInformation() {
-        BacktraceLogger.d(LOG_TAG, "Setting threads information");
-        ThreadData threadData = new ThreadData(report.diagnosticStack);
-        this.mainThread = threadData.getMainThread();
-        this.threadInformationMap = threadData.threadInformation;
-        SourceCodeData sourceCodeData = new SourceCodeData(report.diagnosticStack);
-        this.sourceCode = sourceCodeData.data.isEmpty() ? null : sourceCodeData.data;
+    public String getAgentVersion() {
+        return agentVersion;
+    }
+
+    public Map<String, String> getAttributes() {
+        return attributes;
+    }
+
+    public String getMainThread() {
+        return mainThread;
+    }
+
+    public String[] getClassifiers() {
+        return classifiers;
+    }
+
+    public Map<String, Object> getAnnotations() {
+        return annotations;
+    }
+
+    public Map<String, SourceCode> getSourceCode() {
+        return sourceCode;
+    }
+
+    public BacktraceReport getReport() {
+        return report;
+    }
+
+
+    public static class Builder {
+        private final BacktraceReport report;
+
+        private final String symbolication;
+
+        private String uuid;
+
+        private long timestamp;
+
+        private String[] classifiers;
+
+        private String langVersion;
+
+        private String agentVersion;
+
+        private Map<String, Object> annotations;
+        private Map<String, SourceCode> sourceCode;
+        private Map<String, ThreadInformation> threadInformationMap;
+        private Map<String, String> attributes;
+        private String mainThread;
+
+        public Builder(Context context, BacktraceReport report, Map<String, Object>
+                clientAttributes) {
+            this(context, report, "", clientAttributes);
+        }
+        public Builder(Context context, BacktraceReport report, String symbolication, Map<String, Object>
```

Comment:
This builder seems to me to be a constructor with extra steps.
Why not accept the parameters in methods, e.g. `useContext`, `useSymbolication`, `useProguard`?

Reply:
Improved